source/agora/consensus/EnrollmentManager.d

```diff
@@ -420,6 +420,37 @@ public class EnrollmentManager
         return this.validator_set.count();
     }
 
+    /***************************************************************************
+
+        Check if an enrollment is a valid candidate for the proposed height
+
+        Params:
+            enroll = The enrollment of the target to be checked
+            height = the height of proposed block
+            findUTXO = delegate to find the referenced unspent UTXOs with
+
+        Returns:
+            `null` if the enrollment can be a validator at the proposed height,
+            otherwise a string explaining the reason it is invalid.
+
+    ***************************************************************************/
+
+    public string isValidCandidate (const ref Enrollment enroll, Height height,
+        UTXOFinder findUTXO) @trusted nothrow
+    {
+        if (auto fail_reason = enroll.isInvalidReason(findUTXO))
+            return fail_reason;
+
+        const enrolled = this.validator_set.getEnrolledHeight(enroll.utxo_key);
```

Comment:
```suggestion
        const Height enrolled = this.validator_set.getEnrolledHeight(enroll.utxo_key);
```

Reply:
The format like "`const` {variable name}` is used in many places and we can catch the type easily in this code. So it's not bad I guess. Thanks.